src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/MutableObjectModelBinder.cs

```diff
@@ -186,12 +251,10 @@ private bool CanValueBindAnyModelProperties(MutableObjectBinderContext context)
                 }
             }
 
-            if (!isAnyPropertyEnabledForValueProviderBasedBinding)
+            if (hasBindableProperty && !isAnyPropertyEnabledForValueProviderBasedBinding)
             {
-                // Either there are no properties or all the properties are marked as
-                // a non value provider based marker.
-                // This would be the case when the model has all its properties annotated with
-                // a IBinderMetadata. We want to be able to create such a model.
+                // All the properties are marked as a non value provider based marker like [FromHeader] or
```

Comment:
Confusing me here. Do you mean "are marked _with_ a non value provider based marker"? Something else?